src/System.Management.Automation/engine/remoting/commands/remotingcommandutil.cs

```diff
@@ -170,6 +164,61 @@ internal static void CheckHostRemotingPrerequisites()
                 throw new InvalidOperationException(errorRecord.ToString());
             }
         }
+
+        internal static void CheckPSVersion(Version version)
+        {
+            // PSVersion value can only be 2.0, 3.0, 4.0, 5.0, or 5.1
+            if (version != null)
+            {
+                // PSVersion value can only be 2.0, 3.0, 4.0, 5.0, or 5.1
+                if (!(version.Major >= 2 && version.Major <= 4 && version.Minor == 0) &&
+                    !(version.Major == 5 && version.Minor <= 1))
+                {
+                    throw new ArgumentException(
+                       StringUtil.Format(RemotingErrorIdStrings.PSVersionParameterOutOfRange, version, "PSVersion"));
+                }
+            }
+        }
+
+        /// <summary>
+        /// Checks if the specified version of PowerShell is installed
+        /// </summary>
+        /// <param name="version"></param>
+        internal static void CheckIfPowerShellVersionIsInstalled(Version version)
+        {
+            // Check if PowerShell 2.0 is installed
+            if (version != null && version.Major == 2)
+            {
+#if CORECLR
+                // PowerShell 2.0 is not available for CoreCLR
+                throw new ArgumentException(
+                    PSRemotingErrorInvariants.FormatResourceString(
+                        RemotingErrorIdStrings.PowerShellNotInstalled,
+                        version, "PSVersion"));
+#else
+                // Because of app-compat issues, in Win8, we will have PS 2.0 installed by default but not .NET 2.0
+                // In such a case, it is not enough if we check just PowerShell registry keys. We also need to check if .NET 2.0 is installed.
+                try
```

Comment:
Seems we can remove the code after "else".

Reply:
And if it gets removed, this method will just throw when given v2 and do nothing otherwise. I assume that's all the method should do for PowerShell 6?

Reply:
Hmm, throwing that exception in `#if CORECLR` was a decision when porting the code to NanoServer targeting .NET Core 1.0. Back then, PSv2 was definitely not available on NanoServer, which the only place where PowerShell Core edition is available, so throwing the exception had no problem. Now PowerShell Core is on the desktop, so I'm not sure if we should just throw the exception there (win 10 has removed PSv2, but downlevel OS still has it as an optional windows feature). Besides, if we decide to go with the exception, we probably need to change the error message.

I have updated #3565 to keep track of this cleanup work.

Reply:
Is this the behaviour designed to support `powershell.exe -version 2`?

Reply:
@rjmholt I didn't look into why the PSv2 is taken into consideration here, but possibly related to the scenario where the remote server is PSv2.

Reply:
Yes that makes sense! Well I suppose the behaviour without the #ifdef makes sense then -- no-op unless trying to remote into v2, in which case, **exception**.

Reply:
I believe we don't support PSv2 at all.

Reply:
Yes -- so I'm guessing an exception makes sense then? Something that is red and tells the user we don't support this, but doesn't crash the shell

Reply:
Seems the method is used to run jobs. I guess the remoting (with both local and remote hosts) with PSv2 doesn't work at all.
/cc @PaulHigin Could you please comment too?